cmd/client_test.go

```diff
@@ -95,59 +101,67 @@ func generateRedisProbeDef(name string) *probes_configpb.ProbeDef {
-		if err != nil {
-			t.Errorf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
+	for i := 0; i < testRunCount; i++ {
+		probe := fmt.Sprintf("%s%d", "testExtension", i)
+		if err := client.RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), testHermes.Ctx, generateRedisProbeDef(probe), &myprobe.Probe{}); err != nil {
+			t.Fatalf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
 		}
 
 		// Check if the probe was added correctly
-		listProbesResp, err := client.ListProbes()
+		listProbesResp, err := client.ListProbes(testHermes.Ctx)
 		if err != nil {
-			t.Errorf("ListProbes() failed, expected error: %v, got: %v", nil, err)
+			t.Fatalf("ListProbes() failed, expected error: %v, got: %v", nil, err)
 		}
 
 		probesList := listProbesResp.GetProbe()
-		if probesList[i].GetName() != testProbes[i] {
-			t.Errorf("Expected probe %s to be running, got probe %s running", testProbes[i], probesList[i].GetName())
+
+		// Sorts the probes by their extension number as they are in the format:
+		// "testExtension" + <number>, e.g. "testExtension0".
+		sort.Slice(probesList[:], func(i, j int) bool {
+			probeNum0, _ := strconv.Atoi(probesList[i].GetName()[13:])
+			probeNum1, _ := strconv.Atoi(probesList[j].GetName()[13:])
+			return probeNum0 < probeNum1
+		})
+
+		if probesList[i].GetName() != probe {
+			t.Errorf("Expected probe %s to be running, got probe %s running", probe, probesList[i].GetName())
 		}
 	}
 }
 
 // TestRemoveProbe tests that the RemoveProbe() method removes a probe from Cloudprober
 func TestRemoveProbe(t *testing.T) {
-	testHermes, client := setupCloudproberAndClient()
-	defer teardownCloudproberAndClient(testHermes, client)
+	testHermes, client := setupCloudproberAndClient(t)
+	defer teardownCloudproberAndClient(t, testHermes, client)
 
-	for i := 0; i < len(testProbes); i++ {
-		err := client.RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), testHermes.Ctx, generateRedisProbeDef(testProbes[i]), &myprobe.Probe{})
-		if err != nil {
-			t.Errorf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
+	for i := 0; i < testRunCount; i++ {
+		probe := fmt.Sprintf("%s%d", "testExtension", i)
+		if err := client.RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), testHermes.Ctx, generateRedisProbeDef(probe), &myprobe.Probe{}); err != nil {
+			t.Fatalf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
 		}
 
 		// Check if the probe was added correctly
-		listProbesResp, err := client.ListProbes()
+		listProbesResp, err := client.ListProbes(testHermes.Ctx)
 		if err != nil {
-			t.Errorf("ListProbes() failed, expected error: %v, got: %v", nil, err)
+			t.Fatalf("ListProbes() failed, expected error: %v, got: %v", nil, err)
```

Comment:
Does this need to be fatal, or could you use `t.Errorf` and then complete the rest of your tests?

Here and elsewhere.

Reply:
I previously had it as `t.Errorf`, but Balazs requested that it be changed to `t.Fatalf` in the last [code review,](https://github.com/googleinterns/step224-2020/pull/36#discussion_r482942364) as this is the convention for setup failures.

Reply:
I checked with @leczb and he agreed that it is recommended to use `t.Errorf()` in test functions, and `t.Fataf()` in setup functions.

Reply:
I have made the required changes here and in other test files.